static/js/admin.js

```diff
@@ -657,6 +657,15 @@ function _setup_page() {
                         ui.report_success(i18n.t("waiting period threshold changed!"), waiting_period_threshold_status);
                     }
                 }
+                // Check if no changes made
+                var no_changes_made = true;
+                for(var key in response_data) {
+                    if(['msg', 'result', 'authentication_methods'].indexOf(key) < 0)
+                        no_changes_made = false;
+                }
```

Comment:
Turns out the `authentication_methods` thing is indeed a bug, which I just fixed in e7c9a5087caf7938379bbf0068ef8976fbae45fd; if you rebase, you should be able to drop `authentication_methods` from this list.  The whitespace issues you can see in the Zulip linter.

Otherwise, this piece looks good!